src/Analyzers/Core/Analyzers/UseAutoProperty/AbstractUseAutoPropertyAnalyzer.cs

```diff
@@ -2,315 +2,247 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System;
 using System.Collections.Generic;
 using System.Collections.Immutable;
 using System.Threading;
-using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.CodeStyle;
 using Microsoft.CodeAnalysis.Diagnostics;
 using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.UseAutoProperty
 {
     internal abstract class AbstractUseAutoPropertyAnalyzer<
+        TSyntaxKind,
         TPropertyDeclaration,
         TFieldDeclaration,
         TVariableDeclarator,
         TExpression> : AbstractBuiltInCodeStyleDiagnosticAnalyzer
+        where TSyntaxKind : struct, Enum
         where TPropertyDeclaration : SyntaxNode
         where TFieldDeclaration : SyntaxNode
         where TVariableDeclarator : SyntaxNode
         where TExpression : SyntaxNode
     {
-        private static readonly LocalizableString s_title =
-            new LocalizableResourceString(nameof(AnalyzersResources.Use_auto_property),
-                AnalyzersResources.ResourceManager, typeof(AnalyzersResources));
-
         protected AbstractUseAutoPropertyAnalyzer()
-            : base(IDEDiagnosticIds.UseAutoPropertyDiagnosticId, EnforceOnBuildValues.UseAutoProperty, CodeStyleOptions2.PreferAutoProperties, s_title, s_title)
+            : base(IDEDiagnosticIds.UseAutoPropertyDiagnosticId,
+                   EnforceOnBuildValues.UseAutoProperty,
+                   CodeStyleOptions2.PreferAutoProperties,
+                   new LocalizableResourceString(nameof(AnalyzersResources.Use_auto_property), AnalyzersResources.ResourceManager, typeof(AnalyzersResources)),
+                   new LocalizableResourceString(nameof(AnalyzersResources.Use_auto_property), AnalyzersResources.ResourceManager, typeof(AnalyzersResources)))
         {
         }
 
-        public override DiagnosticAnalyzerCategory GetAnalyzerCategory() => DiagnosticAnalyzerCategory.SemanticSpanAnalysis;
+        /// <summary>
+        /// A method body edit anywhere in a type will force us to reanalyze the whole type.
+        /// </summary>
+        /// <returns></returns>
+        public override DiagnosticAnalyzerCategory GetAnalyzerCategory()
+            => DiagnosticAnalyzerCategory.SemanticDocumentAnalysis;
```

Comment:
@mavasani not sure if this is correct. we def should not be a semanticspan analyzer though as an edit in any method would need to renanalyze the whole type again.

Reply:
This should be fine. We will very soon remove this method invocation from lightbulb path, and hopefully soon after remove this API and enum completely when we remove incremental member edit analysis.